src/language-handlebars/printer-glimmer.js

```diff
@@ -115,11 +115,13 @@ function print(path, options, print) {
       );
     }
     case "MustacheStatement": {
-      const shouldBreakOpeningMustache = isParentOfSomeType(path, [
+      const isParentOfSpecifiedTypes = isParentOfSomeType(path, [
         "AttrNode",
         "ConcatStatement",
         "ElementNode",
       ]);
+      const shouldBreakOpeningMustache =
+        isParentOfSpecifiedTypes && doesNotHaveHashParams(n);
```

Comment:
hi, thanks for picking up this issue. Can you help confirm if this case in [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuExgEIAEB3ATtAc0wgFcYAHMzAX2oB0oAeAEwEsA3APgc0zQE9WcADbNMACgAWAQwDOknr0wxWMYXAC84yAFty0BDGWr1ASkW9Jcaczi4tu-VEOYrNu+ahLMw6VAIk0gSa2hB6BrA+fgFBcJ68prQMjAD0bFwMDGjocAAe5HBg8GKkFFSsULKstpgsHJxJUAJCoopScgpeSipqIY4RRj1mFq7Wtvah4c6RbuPxSr7+gcEOYU4uizHB8560IAA0IBDkKtCyyKDSuPjYAApXCOco0sLY0vznhwBGuNJgANZwGAAZXIfwqBGQMFwJDgh0kMB0wgA6pJVHBZGCwHBgY9VBxVPxkOBZJ8QBVZHYYLdfgQdNJkAAzF6Uw4AK1kuQAQr8AUDgdIdHAADIVOBMllwkAc3LAiHqACKJAg8AlwlZIDBuEpuGJBGErB0Qt1h3IuAqMGR1RgkmQAA4AAym-CU5G-cjEs0YuzscWHACOyvgNOOTxAcgAtM44LZmAcQLg4IHWImaUF6WqNZSdKwoTCpVV-Iqg+KkMz1VKYNIvlbmDbkAAmQ7Q6SsA3+ADCYQzKAxAFZ4yRKQAVatPcsa9iwgCSUFssGBYHNJwAgnPgTB+OpM3BaEA) is fixed by your PR? It should not break opening mustache with a yield & hash. thank you.

Reply:
You're right the yield was also affected with this bug. I've created a fix - do you know if it can be solved in a better way?

Reply:
thanks for the fix. I'm not sure if there's a cleaner way. Your fix lgtm.